johnzon-core/src/test/java/org/apache/johnzon/core/JsonPointerTest.java

```diff
@@ -456,6 +456,19 @@ public void testRemoveArrayElement() {
         assertEquals("[[\"bar\",\"baz\"]]", result.toString()); // [["bar","baz"]]
     }
 
+    @Test
+    public void testRemoveLastArrayElement() {
```

Comment:
maybe add a test for "" too since this case is handled differently

Reply:
If I get your question correct, there is already a test for it: https://github.com/apache/johnzon/blob/d1c6a2f62ef0bb5a798161799eae4602e20a233e/johnzon-core/src/test/java/org/apache/johnzon/core/JsonPointerTest.java#L390-L396

Reply:
I was more thinking about the + case (know it is not what you changed but your change moves from (a || b) to (a || (b && c)), I just want to prevent the fact it becomes ((a || b) && c) 
 - or something else - during another enhancement by a test executed in the build.
Hope it makes sense.